fake-koji/src/main/java/org/fakekoji/jobmanager/JenkinsJobUpdater.java

```diff
@@ -384,15 +386,45 @@ public JobUpdateResults bump(final Set<Tuple<Job, Job>> jobTuples) {
             LOGGER.info("Bumping job " + originalName + " to " + transformedName);
             LOGGER.info("Moving directory " + originalDir.getAbsolutePath() + " to " + transformedDir.getAbsolutePath());
-            Utils.moveDir(originalDir, transformedDir);
+            Utils.moveDir(originalDir, transformedDir); //if exception is thrown from here, better to die with it
+            IOException deleteJobException = null;
+            try {
+                JenkinsCliWrapper.getCli().deleteJobs(originalName).throwIfNecessary();
+            } catch (IOException ex) {
+                LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+                deleteJobException = ex;
+            }
             final File jobConfig = Paths.get(transformedDir.getAbsolutePath(), JENKINS_JOB_CONFIG_FILE).toFile();
             LOGGER.info("Rewriting config of " + transformedName);
-            // FIXME: this won't work, need to call delete(old) and create(new)
-            return new PrimaryExceptionThrower<>(
-                    () -> Utils.writeToFile(jobConfig, transformed.generateTemplate()),
-                    () -> updateManuallyUpdatedJob(transformedName),
-                    new JobUpdateResult(transformedName, true, "bumped from " + originalName)
-            ).call();
+            IOException newCfgException = null;
+            try {
+                Utils.writeToFile(jobConfig, transformed.generateTemplate());
+            } catch (IOException ex) {
+                LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+                newCfgException = ex;
+            }
+            IOException createJobException = null;
+            try {
+                JenkinsCliWrapper.getCli().createManuallyUploadedJob(jobsRoot.getAbsoluteFile(),transformedName).throwIfNecessary();
+            } catch (IOException ex) {
+                LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+                createJobException = ex;
+            }
+            if (deleteJobException == null && newCfgException == null && createJobException == null) {
+                return new JobUpdateResult(transformedName, true, "bumped from " + originalName + " to " + transformedName);
+            } else {
+                String s = "Exception(s) on the fly:";
+                if (deleteJobException != null) {
+                    s = s + " 1) deleteJobException " + deleteJobException.getMessage();
+                }
+                if (newCfgException != null) {
+                    s = s + " 2) newCfgException " + newCfgException.getMessage();
+                }
+                if (createJobException != null) {
+                    s = s + " 3) createJobException  " + createJobException.getMessage();
+                }
+                return new JobUpdateResult(transformedName, false, "bump from " + originalName + " to " + transformedName + " failed. See logs. " + s);
```

Comment:
line too long

Reply:
long lines are good. we are living in 2020
I love terminal. But enforcements to line length when all of us have teo HDD monitors... sorryy....